daemon/graphdriver/overlay2/overlay.go

```diff
@@ -409,13 +409,43 @@ func (d *Driver) Get(id string, mountLabel string) (s string, err error) {
 	}()
 
 	workDir := path.Join(dir, "work")
-	opts := fmt.Sprintf("lowerdir=%s,upperdir=%s,workdir=%s", string(lowers), path.Join(id, "diff"), path.Join(id, "work"))
-	mountLabel = label.FormatMountLabel(opts, mountLabel)
-	if len(mountLabel) > syscall.Getpagesize() {
+
+	var mount func(string, string, string, uintptr, string) error
+	var mountTarget string
+	pageSize := syscall.Getpagesize()
+	splitLowers := strings.Split(string(lowers), ":")
+
+	// Check to see if absolute paths could be used to avoid forking to
+	// mount. Since the entire mount label must fit inside a single page,
+	// to use absolute paths the length of the driver root directory path
+	// must be considered and accounted for each lower layer. In addition
+	// to the root directory, the length of the link file, link directory
+	// name and path separator must be accounted for. The absolute
+	// directories for work and diff are also accounted for along with a
+	// buffer for other mount labels.
+	if (len(d.home)+idLength+len(linkDir)+2)*len(splitLowers)+len(dir)*2+512 < pageSize {
```

Comment:
I had considered just calculating the opts using absolute paths optimistically, I think this might make even more sense considering I found the threshold to be over 60 layers.